### ISSUE TITLE
igraph::graph.data.frame overrides simplegraph:::graph.data.frame

### ISSUE THREAD

**Issue by ugroempi** [gaborcsardi/sankey#1]
even after changing the code to `sankey::graph(...)` in `make_sankey`. 
This gives me the error 
` Error in make_empty_graph(n = 0, directed = directed) : 
  (list) object cannot be coerced to type 'logical' `
that was reported earlier on the other branch.

Best, Ulrike

**Comment by gaborcsardi** [comment]
I am sorry, I am not sure what the problem is here. Is it simply a name clash?

**Comment by ugroempi** [comment]
I don't know. 
`make_sankey` obviously wants to call `sankey::graph` (and implicitly `sankey:::graph.data.frame`) but does in fact call `igraph::graph.data.frame` (igraph apparently exports the method and not only the generic). For that function, the inputs are wrong; as a result `make_sankey` does not work, when igraph is loaded (maybe there are further relevant conditions(?)).

I would have expected importing `graph` from simplegraph, like sankey does, would lead to also using the method from simplegraph, but that seems to be not the case. One solution would be to also export `graph.data.frame` from simplegraph and directly import it in sankey. Or not to export `graph.data.frame` from igraph (but you presumably had a reason for that). Anyway, I think that it would be useful for sankey to work even if igraph is loaded.

Best, Ulrike

**Comment by gaborcsardi** [comment]
Wait, so you are saying that `make_sankey` fails if `igraph` is attached?

**Comment by ugroempi** [comment]
Yes, it seems so. I got the idea for the root cause here: [https://github.com/MangoTheCat/sankey/issues/17](https://github.com/MangoTheCat/sankey/issues/17), and it seems to be the same thing.
